core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java

```diff
@@ -1829,6 +1830,32 @@ public void testRemoveSnapshotsNoOp() throws Exception {
         .isSameAs(current);
   }
 
+  @TestTemplate
+  public void testFileCleanupOnAllRefsAgedOff() {
+    table.newAppend().appendFile(FILE_A).commit();
+    Set<String> expectedDeleteFiles =
+        ImmutableSet.of(table.currentSnapshot().manifestListLocation());
+    String tag = "tag";
+    long tagAgeMs = 50;
+    table
+        .manageSnapshots()
+        .createTag(tag, table.currentSnapshot().snapshotId())
+        .setMaxRefAgeMs(tag, tagAgeMs)
+        .commit();
+    long currentTime = System.currentTimeMillis();
+    table.newAppend().appendFile(FILE_B).appendFile(FILE_C).commit();
+
+    waitUntilAfter(currentTime + tagAgeMs + 10);
```

Comment:
Why +10? `waitUntilAfter` should handle it.

Reply:
It's not needed, I cleaned this up